samples/features/sql-big-data-cluster/connectivity/webhdfs-java-client/src/main/java/com/microsoft/mssql/App.java

```diff
@@ -0,0 +1,118 @@
+                        // First request to get the location in data nodes
+                        try (CloseableHttpResponse response = client2.execute(request)) { 
+                                System.out.println("===============");
+                                String newlocation = response.getFirstHeader("Location").getValue();
+                                HttpEntity entity = response.getEntity();
+                                System.out.println("----------------------------------------");
+                                System.out.println(response.getStatusLine());
+                                System.out.println("----------------------------------------");
+                                if (entity != null) {
+                                        System.out.println(EntityUtils.toString(entity));
+                                }
+                                System.out.println("----------------------------------------");
+                                EntityUtils.consume(entity);
+                                // Second request to put the content to that location
+                                File testUploadFile = new File(inputFilePath);
+                                HttpEntity putData = MultipartEntityBuilder.create().addBinaryBody("upfile", testUploadFile).build();
+                                HttpUriRequest putRequest = RequestBuilder.put(newlocation).setEntity(putData).build();//setEntity(postData).build();
+                                System.out.println("Executing request " + putRequest.getRequestLine());
+                                CloseableHttpResponse response2 = client2.execute(putRequest);
+                                HttpEntity entity2 = response2.getEntity();
+                                System.out.println("----------------------------------------");
+                                System.out.println(response2.getStatusLine());
+                                System.out.println("----------------------------------------");
+                                if (entity2 != null) {
+                                        System.out.println(EntityUtils.toString(entity2));
+                                }
+                                System.out.println("----------------------------------------");
+                                
+                        }
+                } catch (Exception e) {
+                	System.out.println(e);
+                }
+	
+        }
+ 
+	public static void main(String[] args) throws Exception {
+		// These properties can be specified using command line as well by using 
+		// -Djava.security.auth.login.conf=login.conf -Djava.security.krb5.realm=MYDOMAIN.LOCAL
+		// In case command line is used to pass these arguments, then code below setting system properties should not be used..
+		System.setProperty("java.security.auth.login.config", LOGIN_CONF_PATH); // The login conf for Java GSS to use. See readme for a sample.
```

Comment:
For customer to be able to use this out of the box, you could remove these code lines with System.setProperty and in the instructions mention that the user has to run with `java -Djava.security.auth.login.config=login.conf -Djava.security.krb5.realm=aris.local -Djava.security.krb5.kdc=domain.aris.local -Djavax.security.auth.useSubjectCredsOnly=false`
Any user will have to modify this source code in current condition. This should be parameterized to prevent source modification..